fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -471,11 +471,15 @@ def download(app_id, dir = nil)
     # Uploads the modified package back to App Store Connect
     # @param app_id [Integer] The unique App ID
     # @param dir [String] the path in which the package file is located
+    # @param package_path [String] the path to the package file (used instead of app_id and dir)
     # @return (Bool) True if everything worked fine
     # @raise [Deliver::TransporterTransferError] when something went wrong
     #   when transferring
-    def upload(app_id, dir)
-      actual_dir = File.join(dir, "#{app_id}.itmsp")
+    def upload(app_id = nil, dir = nil, package_path: nil)
+      raise "app_id and dir are required or package_path is required" if (app_id.nil? || dir.nil?) && package_path.nil?
+
+      actual_dir = package_path
+      actual_dir ||= File.join(dir, "#{app_id}.itmsp")
```

Comment:
`actual_dir = package_path || File.join(dir, "#{app_id}.itmsp")` ?

Reply:
Yeha yeah yeah, I can change that up!